compiler/qsc/src/interpret.rs

```diff
@@ -454,13 +454,57 @@ impl Interpreter {
     fn lower(
```

Comment:
IMO we really need tests at this level that exercise the FIR pass errors.

Reply:
Added python test for it, will add Rust test shortly.